var/spack/repos/builtin/packages/r3d/package.py

```diff
@@ -0,0 +1,45 @@
+# Copyright 2013-2018 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class R3d(MakefilePackage):
+    """Fast, robust polyhedral intersections, analytic integration, and
+    conservative voxelization."""
+
+    homepage = "https://github.com/devonmpowell/r3d"
+    url      = "https://github.com/devonmpowell/r3d.git"
+
+    version('181219', git=url, commit='47308f68c782ed3227d3dab1eff24d41f6421f21')
+
+    variant("test",  default=False, description="Build R3D regression tests")
+
+    def install(self, spec, prefix):
+
+        make_args = [
+            'CC={0}'.format(spack_cc),
+        ]
+
+        make('libr3d.a', *make_args)
+
+        # R3D does not have an install target so create our own here.
+        mkdirp(prefix.include)
+        _my_headers = find('.', '*.h', recursive=False)
+        for _my_header in _my_headers:
+            install(_my_header, prefix.include)
+        mkdirp(prefix.lib)
+        install('libr3d.a', prefix.lib)
+
+        if '+test' in spec:
+
+            with working_dir('tests'):
+                make('all', *make_args)
+
+                # R3D does not have an install target so create our own here.
+                mkdirp(prefix.test)
+                install('r2d_unit_tests', prefix.test)
+                install('r3d_unit_tests', prefix.test)
+                install('rNd_unit_tests', prefix.test)
```

Comment:
Tell me more about these tests? Are they run during installation? You could also use the `--run-test=root` flag to determine whether or not tests are requested, but since you actually install something, it's prob best to use a variant like you did.

Reply:
These are little regression test binaries that run, standalone, against the built library.  They may be useful to a user for testing the library but not needed at all for general use.